R/importNGWMN_wml2.R

```diff
@@ -147,3 +111,68 @@ importNGWMN_wml2 <- function(input, asDateTime=FALSE, tz=""){
   }
   return(mergedDF)
 }
+
+#' parse the timeseries portion of a waterML2 file
+#' 
+#' Returns data frame columns of all information with each time series measurement;
+#' Anything defined as a default, is returned as an attribute of that data frame.
+#' 
+#' @param input XML with only the wml2:MeasurementTimeseries node and children
+#' @importFrom xml2 xml_attr xml_find_all xml_text 
+#' @importFrom dplyr mutate
+#' @importFrom lubridate parse_date_time
+#' @export
+parseWaterML2Timeseries <- function(input, asDateTime) {
+  gmlID <- xml_attr(input,"id") #TODO: make this an attribute
+  TVP <- xml_find_all(input, ".//wml2:MeasurementTVP")#time-value pairs
+  rawTime <- xml_text(xml_find_all(TVP,".//wml2:time"))
+  
+  valueNodes <- xml_find_all(TVP,".//wml2:value")
+  values <- as.numeric(xml_text(valueNodes))
+  nVals <- length(values)
+  
+  #df of date, time, dateTime
+  oneCol <- rep(NA, nVals) 
+  timeDF <- data.frame(date=oneCol, time=oneCol, dateTime=oneCol)
+  splitTime <- data.frame(matrix(unlist(strsplit(rawTime, "T")), nrow=nVals, byrow = TRUE), stringsAsFactors=FALSE)
+  if(ncol(splitTime) > 1){ #some sites only have a date
+    names(splitTime) <- c("date", "time")
+  }else{
+    names(splitTime) <- "date"
+    splitTime <- mutate(splitTime, time = NA)
+  }
+  
+  timeDF <- mutate(splitTime, dateTime = NA)
+  logicVec <- nchar(rawTime) > 19
+  if(!all(!logicVec)) { #otherwise sets it to char <NA>
+    timeDF$dateTime[logicVec] <- rawTime[logicVec]
+  }
+  if(asDateTime){
+    timeDF$dateTime <- parse_date_time(timeDF$dateTime, c("%Y","%Y-%m-%d","%Y-%m-%dT%H:%M","%Y-%m-%dT%H:%M:%S",
+                                                          "%Y-%m-%dT%H:%M:%OS","%Y-%m-%dT%H:%M:%OS%z"), exact = TRUE)
+    #^^setting tz in as.POSIXct just sets the attribute, does not convert the time!
+    attr(time, 'tzone') <- tz
```

Comment:
Weird, apparently that hasn't caused an error because there is a `tz` function in lubridate that hides the fact the argument is missing.